core/task/manager.go

```diff
@@ -695,14 +704,20 @@ func (m *Manager) KillTasks(taskIds []string) (killed Tasks, running Tasks, err
 	}
 
 	for _, id := range toKill.GetTaskIds() {
-		m.ackKilledTasks[id] = make(chan struct{})
+		m.ackKilledTasks.addAckChannel(id)
 	}
 
 	killed, running, err = m.doKillTasks(toKill)
 	for _, id := range killed.GetTaskIds() {
-		<- m.ackKilledTasks[id]
-		close(m.ackKilledTasks[id])
-		delete(m.ackKilledTasks, id)
+		ack,_ := m.ackKilledTasks.getValue(id)
+		<- ack
+		close(ack)
+		m.ackKilledTasks.deleteKey(id)
```

Comment:
Was there a race for the ackKilledTasks structure?

Reply:
I received a `WARNING DATA RACE`, so I thought it will be safer to fix this.